pkg/engine/validation.go

```diff
@@ -172,59 +169,69 @@ func addRuleResponse(log logr.Logger, resp *response.EngineResponse, ruleResp *r
 
 type validator struct {
 	log              logr.Logger
-	ctx              *PolicyContext
+	policyContext    *PolicyContext
 	rule             *kyvernov1.Rule
 	contextEntries   []kyvernov1.ContextEntry
 	anyAllConditions apiextensions.JSON
 	pattern          apiextensions.JSON
 	anyPattern       apiextensions.JSON
 	deny             *kyvernov1.Deny
 	podSecurity      *kyvernov1.PodSecurity
+	foreach          []kyvernov1.ForEachValidation
 	rclient          registryclient.Client
+	nesting          int
```

Comment:
What is nesting actually used for? It seems to just be in the context - so only in logs? Or am I missing something

Reply:
Hi @MarcelMue - the nesting count is used to calculate `elementIndex` in the context. This way an expression like `/spec/tls/{{elementIndex0}}/hosts/{{elementIndex1}}` can be resolved.